url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/Hipo.java

```diff
@@ -0,0 +1,1790 @@
+            if (dispara) {
+                //Obtenemos el consecuente
+                ArrayList<Elemento> consecuente = r.getConsecuente().getElementos();
+                for (Elemento e : consecuente) {
+                    //El elemento que indica el consecuente tendrá el valor que indica el consecuente salvo que ya esté escrito por una regla anterior (y mejor en teoría)
+                    //Indicaremos el valor como la probabilidad de que sea 1 en función de la confianza de la regla
+                    //Comprobamos que nadie haya escrito ya la respuesta
+                    float valorActual = respuesta[e.getSubindice()][e.getEntrada()];
+                    if (valorActual != 1 && valorActual != 0) {
+                        int valorNuevo = e.isVerdadero() ? 1 : 0;
+                        //respuesta[e.getSubindice()][e.getEntrada()] = valorNuevo;
+                        //Si es cero la probabilidad se invierte
+                        float probabilidad = (float) r.getConfianza();
+                        //Multiplicamos la probabilidad por la probabilidad de que dispare
+                        //así penalizamos a las reglas que tienen muchos antecedentes inciertos
+                        probabilidad *= probabilidad_disparo;
+                        if (valorNuevo == 0) {
+                            probabilidad = 1 - probabilidad;
+                        }
+
+                        //Sólo la aceptamos si defiende un 0 y es menor que 0.5 o si defiende un 1 y es mayor que 0.5
+                        //ya que las reglas no son dobles implicaciones, es decir, que no se cumpla una regla
+                        //no aporta ninguna información sobre el cumplimiento o no de otras
+                        if (valorNuevo == 0 && probabilidad < 0.5
+                                || valorNuevo == 1 && probabilidad > 0.5) {
+
+                            //Si la probabilidad es mejor que la que ya había entonces se sustituye
+                            //Es mejor cuanto más alejada está de 0.5
+                            //Bondad de la Probabilida actual
+                            float BvalorActual = Math.abs(valorActual - 0.5f); //Calculamos la distancia a 0.5 para cada una
+                            float BvalorNuevo = Math.abs(probabilidad - 0.5f);
+                            if (BvalorNuevo > BvalorActual) {
+                                respuesta[e.getSubindice()][e.getEntrada()] = probabilidad; //Nos quedamos con la que está más alejada del desconocimiento, del 0.5
+                            }
+                        }
+                    }
+                }
+            }
+
+        }
+
+        return respuesta;
+    }
+
+    /**
+     * Devolver la forma más sencilla de explicar lo sucedido, rellena lo que
+     * falta
+     *
+     * @param entradas
+     * @param reglas
+     * @return
+     */
+    private int[][] evaluar_probable(int[][] entradas, Teoria teoria) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.